tracer/probe.go

```diff
@@ -0,0 +1,74 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package tracer // import "go.opentelemetry.io/ebpf-profiler/tracer"
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/link"
+)
+
+type ProbeSpec struct {
+	Type     string
+	Target   string
+	Symbol   string
+	ProgName string
+}
+
+func ParseProbe(spec string) (*ProbeSpec, error) {
+	parts := strings.SplitN(spec, ":", 3)
```

Comment:
As we handle here user input, should we handle the case where capitalized letters are input?

Reply:
no strong opinions here.  `strings.ToLower()` is always a good start and for the rest pkg `link` will return an error if it does not fit.